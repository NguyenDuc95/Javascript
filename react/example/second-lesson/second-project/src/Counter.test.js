import React from 'react';

import renderer from "react-test-renderer";
import Counter from './Component/Counter';
import { mount } from 'enzyme';

describe('Counter', () => {
    it('should match snapshot', () => {
        const component = renderer.create(<Counter match={{ params: { id: 1 } }} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('button text should be increase2', () => {
        const component = mount(<Counter match={{ params: { id: 1 } }} />);
        expect(component.find("button").text()).toEqual("Increase2")
    })
    it('init', () => {
        const component = mount(<Counter match={{ params: { id: 1 } }} />);
        expect(component.find("div.counter").text()).toEqual("1")
    })
})