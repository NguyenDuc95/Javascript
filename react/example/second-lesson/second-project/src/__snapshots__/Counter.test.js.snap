// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="counter"
  >
    1
  </div>
  <button
    onClick={[Function]}
    value="Increase"
  >
    Increase2
  </button>
  <div>
    <select>
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
    </select>
  </div>
</div>
`;
